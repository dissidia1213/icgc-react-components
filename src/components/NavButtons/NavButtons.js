//@flow

import React, { Component } from "react";
import { Button as ButtonSUI } from "semantic-ui-react";
import PropTypes from "prop-types";

//import * as CONSTANTS from "../../utils/constants";
import styles from "./NavButtons.css";

export default class NavButtons extends Component {

	render() {

		return (
			<ButtonSUI.Group className={styles.myNavButtons}>
				<ButtonSUI basic color={this.props.color ? this.props.color : "orange"} disabled={this.props.disabledPrev} floated='left' labelPosition='left' icon='left chevron' content='Anterior' onClick={this.props.goPrevStep}/>
				<ButtonSUI color={this.props.color ? this.props.color : "orange"} disabled={this.props.disabledNext} floated='right' labelPosition='right' icon='right chevron' content='SegÃ¼ent' onClick={this.props.goNextStep}/>
			</ButtonSUI.Group>
		);

	}

}

NavButtons.propTypes = {
	goPrevStep: PropTypes.func,
	goNextStep: PropTypes.func,
	disabledPrev: PropTypes.bool,
	disabledNext: PropTypes.bool,
	color: PropTypes.string
};
