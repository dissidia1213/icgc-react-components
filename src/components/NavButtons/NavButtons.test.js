//@flow

import { shallow } from "enzyme";
import React from "react";
import NavButtons from "./NavButtons";
import "../../setupTests";

it("Expect to ", ()=> {

	const mockProps = {
		color: "teal"
	};

    

    const wrapper = shallow(<NavButtons ></NavButtons>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.length).toEqual(1);
	//expect(wrapper.props().fullWidth).toEqual(true);


});