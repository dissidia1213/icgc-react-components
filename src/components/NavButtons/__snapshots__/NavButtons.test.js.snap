// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavButtons />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Button
          as="button"
          basic={true}
          color="yellow"
          content="Anterior"
          disabled={undefined}
          floated="left"
          icon="left chevron"
          labelPosition="left"
          onClick={undefined}
/>,
        <Button
          as="button"
          color="yellow"
          content="Seg端ent"
          disabled={undefined}
          floated="right"
          icon="right chevron"
          labelPosition="right"
          onClick={undefined}
/>,
      ],
      "className": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": "button",
          "basic": true,
          "color": "yellow",
          "content": "Anterior",
          "disabled": undefined,
          "floated": "left",
          "icon": "left chevron",
          "labelPosition": "left",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": "button",
          "color": "yellow",
          "content": "Seg端ent",
          "disabled": undefined,
          "floated": "right",
          "icon": "right chevron",
          "labelPosition": "right",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Button
            as="button"
            basic={true}
            color="yellow"
            content="Anterior"
            disabled={undefined}
            floated="left"
            icon="left chevron"
            labelPosition="left"
            onClick={undefined}
/>,
          <Button
            as="button"
            color="yellow"
            content="Seg端ent"
            disabled={undefined}
            floated="right"
            icon="right chevron"
            labelPosition="right"
            onClick={undefined}
/>,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "button",
            "basic": true,
            "color": "yellow",
            "content": "Anterior",
            "disabled": undefined,
            "floated": "left",
            "icon": "left chevron",
            "labelPosition": "left",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "button",
            "color": "yellow",
            "content": "Seg端ent",
            "disabled": undefined,
            "floated": "right",
            "icon": "right chevron",
            "labelPosition": "right",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
